keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -1402,19 +1431,11 @@ Below are some examples to consider, in addition to the aforementioned [maturity
 #### GA
 
 - Address reviews and bug reports from Beta users
```

Comment:
documentation for this feature is very weak. I found one paragraph mentioning it. I'd suggest to make the detailed documentaiton with examples to be a beta 2 requirement

Reply:
Thank you for the comment. I’m happy to discuss the missing stuff and update it.  

For the context, the current documentation for this feature includes:
- https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-failure-policy (pod failure policy for Jobs with example configuration using the DisruptionTarget condition)
- https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions (pod disruption conditions added when a pod is deleted, briefly mentioning the transition phase by PodGC, I’m thinking about expanding this paragraph with the kubelet changes).
- https://kubernetes.io/docs/tasks/job/pod-failure-policy/ (example of using the pod failure policy to control handling of pod failures in real life scenarios)

First, I’ve update the PR to add a point to clarify the recent changes (for Beta 2). However, the transitions to the Failed phase seem to be technical fixes, so not sure they require two user-facing examples (they don't require any specific handling in the Job's `.spec.podFailurePolicy`, they just make sure that the pod failure policy is matched against pods in terminal phases), but maybe we could expand the tasks second with another one. Do you think of a particular example to add?